ui/lib/kv/addon/components/page/secret/metadata/version-diff.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { kvDataPath } from 'vault/utils/kv-path';
+
+/**
+ * @module KvSecretMetadataVersionDiff renders the version diff comparison
+ * <Page::Secret::Metadata::VersionDiff
+ *  @metadata={{this.model.metadata}}
+ *  @path={{this.model.path}}
+ *  @backend={{this.model.backend}}
+ *  @breadcrumbs={{this.breadcrumbs}}
+ * />
+ *
+ * @param {model} metadata - Ember data model: 'kv/metadata'
+ * @param {string} path - path to request secret data for selected version
+ * @param {string} backend - kv secret mount to make network request
+ * @param {array} breadcrumbs - Array to generate breadcrumbs, passed to the page header component
+ */
+
+/* eslint-disable no-undef */
+export default class KvSecretMetadataVersionDiff extends Component {
+  @service store;
+  @tracked leftVersion;
+  @tracked rightVersion;
+  @tracked visualDiff;
+  @tracked statesMatch = false;
+
+  constructor() {
+    super(...arguments);
+
+    // initialize with most recently (before current), active version on left
+    const olderVersions = this.args.metadata.sortedVersions.slice(1);
+    const recentlyActive = olderVersions.find((v) => !v.destroyed && !v.isSecretDeleted);
+    this.leftVersion = Number(recentlyActive?.version);
+    this.rightVersion = this.args.metadata.currentVersion;
+
+    // this diff is from older to newer (current) secret data
+    this.createVisualDiff();
+  }
+
+  // diffs with an inactive version only happens on initialization if the current version is inactive
```

Comment:
```suggestion
  // this can only be true on initialization if the current version is inactive
  // selecting a deleted/destroyed version is otherwise disabled
```